### DISCUSSION TITLE
How to deploy for Linux

### DISCUSSION THREAD

**Discussion by Realgigabyte** [unoplatform/uno#8408] [Q&A] [answered]
Hi,

I think I am missing a point here.
I tried to create a UNO app on Visual Studio on Windows.
I can compile for UWP or deploy for Android (.apk)

How do I consume the app on Linux?
I mean how do I release the app and run/install on a Linux machine?

Any help much appreciated.

**Comment by jeromelaban** [comment]
We're probably missing a bit of documentation for that part, but you can build you app on windows (using the `.Skia.Gtk` head) using `dotnet publish`. In the resulting folder, you will be able to bring this to Linux and run `dotnet MyApp.Skia.Gtk.dll`.

There are other methods of deployment, using [self-contained packaging](https://docs.microsoft.com/en-us/dotnet/core/deploying/#publish-self-contained), or [runtime-specific compilation](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-publish#examples) that can be used.

**Comment by KWodarczyk** [comment]
@jeromelaban  can you please update to documentation. I would be a very helpful is there was a guideline describing the flows step by step.  Eg. developing for desktop but publishing and running on linux it's one of those flow.

**Reply by jeromelaban** [reply]
@KWodarczyk we've updated [the documentation](https://platform.uno/docs/articles/uno-publishing-desktop.html) since then, but we still have some updates coming.

**Reply by KWodarczyk** [reply]
So at the moment we have a guideline on how to build the up for publishing (dotnet publish -f net8.0-desktop -c Release -o ./publish) but not how to run it on linux ? Is that correct ?

**Reply by jeromelaban** [reply, marked as answer]
I'm not sure to understand the difference you make here, but maybe you're talking about packaging? Running on linux right now is done by using `dotnet [yourapp].dll`. For extended packaging (e.g. snap or similar), this is a work in progress.

**Reply by mcNets** [reply]
If you deploy a self-contained app, no matter if you directly deploy to the Linux machine:

`dotnet publish -r linux-arm64 -c Release -f net8.0 --self-contained true -o \\$ip\folder /p:PublishSingleFile=false`

or using a share and then copy/paste.

You must give execution permission to the application (not the file with .dll extension)

`sudo chmod +x /folder/MyApp`

and then execute it directly

`./MyApp`

You can do the same by installing dotnet runtime to the Linux machine but, the dotnet version installed and the dotnet version used to build your application must match, for example if you used `8.0.107` to build your application you cannot install `8.0.206` in the Linux machine. Otherwise you only can execute the app using `dontet myApp.dll`

**Comment by lindexi** [comment]
@Realgigabyte And you can use https://github.com/quamotion/dotnet-packaging to help you pack to `.deb` install file and so on.
